design/baremetal-operator/bmh_non-bootable_iso.md

```diff
@@ -0,0 +1,134 @@
+# Add non-bootable iso attach API to BareMetalHost
+
+## Status
+
+Implementable
+
+## Summary
+
+Add a new field in BareMetalHost so that it is possible to attach a generic
+non-bootable ISO (a CD "ISO 9660" image) image via Ironic, after the host has been provisioned.
+
+## Motivation
+
+In certain scenarios it is desirable to attach a non-bootable iso image
+to a system after provisioning has finished, for example :
+
+A configuration ISO can be used to provide first-boot configuration to a host
+deployed from a standard image in advance. An operator may have a pool of
+powered off hosts with operating system already provisioned. Then, adding
+a host to the cluster boils down to powering it on with a configuration ISO provided.
+
+### Goals
+
+Expose the Ironic API to attach non-bootable iso images via Metal3.
+
+### Non-Goals
+
+Supporting non-ISO images i.e. media types other than CD/DVD, supporting
+drivers other than Redfish and its derived drivers.
+
+## Proposal
+
+### BMO Proposal
+
+Add a new optional section in the BareMetalHost spec containing details of the
+non-bootable iso image, example :
+
+#### Current design
+
+```yaml
+  spec:
+    dataImage:
+      url: http://1.2.3.4/image.iso
+```
+
+#### Desired future design
+
+```yaml
+  spec:
+    dataImage:
+      url: http://1.2.3.4/image.iso
+      credentialsName: dataimage-secret
+      disableCertificateVerification: false
+```
+
+Secret holding the image url credentials, base64 encoded :
```

Comment:
nit: this is redundant, we know how to create secrets.

Reply:
removed this part